[RDAPI-28228] EMT image delivery documentation

sgoka-axway opened the pull request:
Thank you for your contribution to the Axway-Open-Docs repo.

## Describe the changes

Updated the how-tp guide for deploying the GW with docker containers to describe new configuration options such as group id and licensing.

Moved the 'Deploy GW with Helm' guide to the 'How-to' section. Added descriptions for new configuration options and image download locations.

## Deploy preview link

http://pr-2471.axway-open-docs.opendocs-builder.pcloud.axway.int/docs/apim_howto_guides/configuring_apigw_container/

_Although Netlify will add this link below, it will redirect to the main page of the documentation, not to an specific page._

## Checklist for contributors

Before submitting this PR, please make sure:

_This is simply a reminder of what we are going to look for before merging your change._

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)
* [x] You have verified that all status checks have passed

sgoka-axway commented:
This is initial draft, not ready for review.

andreamussap commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -1,12 +1,25 @@
 {
-    "title": "Create an API Gateway with Docker volumes",
-    "linkTitle": "Create an API Gateway with Docker volumes",
+    "title": "Deploy API Gateway with Axway Images",
```
(Axway Images) --> sounds a bit abstract... could it be 'Axway Docker images', 'Axway containerized images', 'Axway Helm/chart images' ?

sgoka-axway replied:
Updated to 'Axway Docker images', ty!

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -1,12 +1,35 @@
     "weight": "30",
     "date": "2022-02-01",
-    "description": "Configure an API Gateway Docker or Admin Node Manager container to use a persisted volume for configuration."
+    "description": "A high level introduction to deploy the API Gateway with mounted volumes utilising Axway images or their own API Gateway images. This section details where to download the images and sample configuration files, and how to configure an API Gateway or Admin Node Manager or Analytics docker containers to use a persisted volume for customized configuration."
```
```suggestion
    "description": "A high level introduction to deploy the API Gateway with mounted volumes utilising Axway images or custom-built images. This section details where to download the images and sample configuration files, and how to configure an API Gateway, Admin Node Manager or Analytics docker container to use a persisted volume for customized configuration."
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -1,12 +1,35 @@
+
+### Download Axway docker images
+
+Download Axway docker images  i.e) Admin Node Manager, API Gateway and API Gateway Analytics from [axway.repository.com](http://axway.repository.com/).
```
Remove the ) after i.e

sgoka-axway replied:
Done!

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -1,12 +1,35 @@
+
+By default the images come with no license file. In order to start the containers successfully, the license file must be externalized. See how to externalize a license file using docker run [Build EMT images without a license file](/docs/apim_howto_guides/apigw_in_containers/#build-emt-images-without-a-license-file)
+
+{{< alert title="Note" >}}The helm chart values.yaml has commented out sample license file, provide the valid license file to deploy and start the helm images successfully.
```
```suggestion
{{< alert title="Note" >}}The helm chart values.yaml has commented out sample license file. Paste the contents of a valid license file into the appropriate sections of either the values.yaml file or the override yaml file to deploy and start the helm images successfully.
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -1,12 +1,35 @@
+
+{{< alert title="Note" >}}The helm chart values.yaml has commented out sample license file, provide the valid license file to deploy and start the helm images successfully.
+For Admin Node Manager the license file is optional but is required only when FIPS mode is on.
+For API Gateway and API Gateway Analytics the license file is must to run the pods successfully.{{< /alert >}}
```
instead of 'is must' put 'is a requirement'

sgoka-axway replied:
Removed the helm info from here as we discussed. This will be covered as part of another ticket in helm page updates.

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -196,6 +223,48 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
 
+### Update domain certs at runtime
+
+By default the image contains the default domain certs, to update the domain certs, like other configurations jvm.xml, envSettings.props etc domain certs can be added to the docker container at runtime by way of the /merge/apigateway Docker volume. You must store these configurations locally, in a directory structure which mirrors /opt/Axway/apigateway subfolder structure inside the Docker contai ...
```
```suggestion
By default the image contains the default domain certs. To update the domain certs, like other configurations such as  jvm.xml, envSettings.props etc, domain certs can be added to the docker container at runtime by way of the /merge/apigateway Docker volume. You must store these configurations locally, in a directory structure which mirrors /opt/Axway/apigateway subfolder structure inside the Docker container.
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
````diff
@@ -60,6 +85,8 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
            -e ACCEPT_GENERAL_CONDITIONS=yes analytics:0.0.1
 ```
 
+{{< alert title="Note" >}}By default the Axway docker images doesn't contain the mysql jar, to run the API Gateway Analytics successfully need to externalize the mysql jar.{{< /alert >}}
````
```suggestion
{{< alert title="Note" >}}By default the Axway docker images don't contain the mysql jar. To run the API Gateway Analytics successfully need to externalize the mysql jar.{{< /alert >}}
```

sgoka-axway replied:
Done, ty!

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -16,6 +39,8 @@ The major steps to create and configure your API Gateway or Admin Node Manager i
 
 This process applies to API Gateway or Admin Node Manager configurations only, such as policy management, system property changes, environment settings, logging configurations, and son on.
 
+{{< alert title="Note" >}}The sample configuration files can be retrieved from the running container, copy the sample configuration you want to update from running container to your local machine. For example, Admin Node Manager envSettings.props file is available at apigateway/conf/envSettings.props.Use **docker cp** or **kubectl cp** to copy files to your local machine based on the environment. ...
```
```suggestion
{{< alert title="Note" >}}The sample configuration files can be retrieved from the running container. Copy the sample configuration you want to update from a running container on to your local machine. For example, Admin Node Manager envSettings.props file is available at apigateway/conf/envSettings.props. Use **docker cp** or **kubectl cp** to copy files to your local machine based on the environment. Update the file and mount it using docker volume/persistent storage at runtime.{{< /alert >}}
```

sgoka-axway replied:
Done, removed kubectl cp from here.

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -196,6 +223,48 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
+
+This folder structure is then mounted to the /merge/apigateway directory of the container.
+
+{{< alert title="Note" >}}When updating the domain certs at runtime, the ES_PASSPHRASE and DOMAIN_KEY_PASSPHRASE must be provided if present else takes the default passphrases.{{< /alert >}}
```
```suggestion
{{< alert title="Note" >}}When updating the domain certs at runtime, the ES_PASSPHRASE and DOMAIN_KEY_PASSPHRASE must be provided if present, otherwise it takes the default passphrases.{{< /alert >}}
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -196,6 +223,48 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
+
+{{< alert title="Note" >}}When updating the domain certs at runtime, the ES_PASSPHRASE and DOMAIN_KEY_PASSPHRASE must be provided if present else takes the default passphrases.{{< /alert >}}
+
+During this process the ANM and AGW topology SSL certs gets created and signed by the domain certificates provided at runtime. To secure the communications between the Admin Node Manager and API Gateways, both the SSL certificates should get signed with the same domain certificate else there won't be any communication.
```
```suggestion
During this process the ANM and AGW topology SSL certs get created and signed by the domain certificates provided at runtime. To secure the communications between the Admin Node Manager and API Gateways, both the SSL certificates should be signed with the same domain certificate otherwise there won't be any communication between them.
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -196,6 +223,48 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
+
+### Set Group ID at runtime
+
+Group id can now be set at runtime as an environment variable, all the containers in the group are built with same image, which means contains the same configuration.
```
```suggestion
Group id can now be set at runtime as an environment variable. All the containers in the group are built with same image, which means they contain the same configuration.
```

mrussell-axway commented on content/en/docs/apim_howto_guides/configuring_apigw_container.md:
```diff
@@ -196,6 +223,48 @@ docker run -d -p 8040:8040 --name=analytics --network=api-gateway-domain \
+
+### Restart command for helm
+
+Thi below command is used to restart the pods in the deployment to effect the externalization updates.
```
```suggestion
The command below is used to restart the pods in the deployment to affect the externalization updates.
```

sgoka-axway replied:
Removed the helm info from here as we discussed. This will be covered as part of another ticket in helm page updates.

mrussell-axway reviewed (commented):
Hi Sirisha - great work - just a few syntax errors. Thanks

sgoka-axway commented:
> Hi Sirisha - great work - just a few syntax errors. Thanks

Thanks Mary for the review comments, corrected syntax errors and also updated the doc as we discussed yesterday and removed all helm info from here.
